src/Middleware/Rewrite/src/RewriteMiddleware.cs

```diff
@@ -93,6 +95,16 @@ public Task Invoke(HttpContext context)
                         throw new ArgumentOutOfRangeException($"Invalid rule termination {rewriteContext.Result}");
                 }
             }
+
+            // If a rule changed the path we want routing to find the new endpoint
+            if (originalPath != context.Request.Path)
+            {
+                if (context.GetEndpoint() is not null)
+                {
+                    context.SetEndpoint(null);
+                }
```

Comment:
```suggestion
                context.SetEndpoint(null);
```
How about always setting it null? The code is handling the null feature and also `context.Features.Get<IEndpointFeature>` would be called once.
https://github.com/dotnet/aspnetcore/blob/8b30d862de6c9146f466061d51aa3f1414ee2337/src/Http/Http.Abstractions/src/Routing/EndpointHttpContextExtensions.cs#L55-L61